### PR TITLE
Add support for Get action invocations Workflows (CS2)

### PR DESCRIPTION
**Pull request by Armando-Ibarra**

### CODE REVIEW

File: src/main/java/com/checkout/workflows/four/actions/response/WorkflowInvocationsResponse.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package com.checkout.workflows.four.actions.response;
+
+import com.checkout.common.Resource;
+import com.google.gson.annotations.SerializedName;
+import lombok.Data;
+
+import java.util.List;
+
+@Data
+public class WorkflowInvocationsResponse extends Resource {
```
Comment (rsaestrela) [inline src/main/java/com/checkout/workflows/four/actions/response/WorkflowInvocationsResponse.java:10]: Please rename to `WorkflowActionInvocationsResponse`.

File: src/main/java/com/checkout/workflows/four/actions/response/WorkflowActionType.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package com.checkout.workflows.four.actions.response;
```
Comment (rsaestrela) [inline src/main/java/com/checkout/workflows/four/actions/response/WorkflowActionType.java:1]: I think the right place for this enum is under `com.checkout.workflows.four.actions`, along with `WorkflowActionType`, wdyt?

File: src/main/java/com/checkout/workflows/four/actions/response/WorkflowActionInvocation.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+    private Boolean retry;
+
+    private Boolean succeeded;
+
+    @SerializedName("final")
+    private Boolean finalAttempt;
+
+    @SerializedName("result_details")
+    private String resultDetails;
```
Comment (rsaestrela) [inline src/main/java/com/checkout/workflows/four/actions/response/WorkflowActionInvocation.java:24]: This should be `@SerializedName("result")` and a `Map<String, Object>`.

**Review by rsaestrela — CHANGES_REQUESTED** [review]

File: src/test/java/com/checkout/workflows/four/WorkflowReflowTestIT.java [hunk 4]
```diff
@@ -1,18 +1,19 @@
...
 import org.junit.jupiter.api.Test;
 
 import java.util.Collections;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertNotNull;
 import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 
 class WorkflowReflowTestIT extends AbstractWorkflowTestIT {
```
Comment (rsaestrela) [inline src/test/java/com/checkout/workflows/four/WorkflowReflowTestIT.java:18]: Why this test in `WorkflowReflowTestIT`? Can you please create a different test class for this? For example `WorkflowActionsIntegrationTest`.

File: src/main/java/com/checkout/workflows/four/actions/response/WorkflowActionInvocation.java (hunk 3 above)
Comment (Armando-Ibarra) [reply src/main/java/com/checkout/workflows/four/actions/response/WorkflowActionInvocation.java:24]: It is a Map yes, but the name is result_details
https://api-reference.checkout.com/preview/crusoe/#tag/Workflows/paths/~1workflows~1events~1{eventId}~1actions~1{workflowActionId}/get!c=200&path=action_invocations/result_details&t=response

File: src/main/java/com/checkout/workflows/four/actions/response/WorkflowInvocationsResponse.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    @SerializedName("event_id")
+    private String eventId;
+
+    @SerializedName("workflow_action_id")
+    private String workflowActionId;
+
+    @SerializedName("action_type")
+    private String actionType;
+
+    private WorkflowActionType status;
```
[the remark this answers is not in the document]
Comment (Armando-Ibarra) [reply src/main/java/com/checkout/workflows/four/actions/response/WorkflowInvocationsResponse.java:24]: I just realize we already have WorkflowActionType

**Pull request merged**